module/src/clients/surface_temp.c

```diff
@@ -62,6 +76,7 @@ static int ssam_tmp_get_temperature(struct ssam_device *sdev, u8 iid, long *temp
 struct ssam_temp {
 	struct ssam_device *sdev;
 	s16 sensors;
+	char names[16][sizeof_field(struct ssam_tmp_get_name, name)];
```

Comment:
I 100% tried to do something like `sizeof(struct ssam_tmp_get_name.name)` first... which failed, some searching led me to this [sizeof_field](https://stackoverflow.com/a/42757140) from `linux/stddef.h`.  I was pretty surprised that this is not built in functionality.

Reply:
I think it would be better to define both as macros. Then that construct with `sizeof_field()` won't be needed and it's a bit cleaner to read.

Reply:
Okay, I hope I got this right, I wouldn't call a `define` of a constant a macro, but that may be because I'm not a C programmer :joy: